diff --git a/docker/distro_release/Dockerfile.ubuntu b/docker/distro_release/Dockerfile.ubuntu
index 6648481..6510d5a 100644
--- a/docker/distro_release/Dockerfile.ubuntu
+++ b/docker/distro_release/Dockerfile.ubuntu
@@ -1,11 +1,10 @@
-FROM ubuntu:16.04
+FROM fpco/stack-build:lts-9.6
 
 RUN BUILD_DEPS="curl ca-certificates build-essential" && \
     apt-get -qq update && \
     apt-get -qqy --no-install-recommends install \
     $BUILD_DEPS \
     libpq-dev && \
-    curl -sSL https://get.haskellstack.org/ | sh && \
     apt-get -qq clean && \
     rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
diff --git a/src/PostgREST/Auth.hs b/src/PostgREST/Auth.hs
index 3f29fa4..ce900bc 100644
--- a/src/PostgREST/Auth.hs
+++ b/src/PostgREST/Auth.hs
@@ -56,7 +56,7 @@ jwtClaims secret audience payload =
   Whether a response from jwtClaims contains a role claim
 -}
 containsRole :: JWTAttempt -> Bool
-containsRole (JWTClaims claims) = M.member "role" claims
+containsRole (JWTClaims claims) = M.member "https://kofi.sexy/role" claims
 containsRole _                  = False
 
 {-|
diff --git a/src/PostgREST/Middleware.hs b/src/PostgREST/Middleware.hs
index 8684430..7db5442 100644
--- a/src/PostgREST/Middleware.hs
+++ b/src/PostgREST/Middleware.hs
@@ -40,10 +40,10 @@ runWithClaims conf eClaims app req =
         cookiesSql = map (pgFmtEnvVar "request.cookie.") $ iCookies req
         claimsSql = map (pgFmtEnvVar "request.jwt.claim.") [(c,unquoted v) | (c,v) <- M.toList claimsWithRole]
         setRoleSql = maybeToList $
-          (\r -> "set local role " <> r <> ";") . toS . pgFmtLit . unquoted <$> M.lookup "role" claimsWithRole
+          (\r -> "set local role " <> r <> ";") . toS . pgFmtLit . unquoted <$> M.lookup "https://kofi.sexy/role" claimsWithRole
         setSchemaSql = ["set schema " <> pgFmtLit (configSchema conf) <> ";"] :: [Text]
         -- role claim defaults to anon if not specified in jwt
-        claimsWithRole = M.union claims (M.singleton "role" anon)
+        claimsWithRole = M.union claims (M.singleton "https://kofi.sexy/role" anon)
         anon = String . toS $ configAnonRole conf
         customReqCheck = (\f -> "select " <> toS f <> "();") <$> configReqCheck conf
   where
diff --git a/test/Feature/AudienceJwtSecretSpec.hs b/test/Feature/AudienceJwtSecretSpec.hs
index 35be4dd..efe9048 100644
--- a/test/Feature/AudienceJwtSecretSpec.hs
+++ b/test/Feature/AudienceJwtSecretSpec.hs
@@ -20,13 +20,13 @@ spec = describe "test handling of aud claims in JWT" $ do
 
         {
           "exp": 9999999999,
-          "role": "postgrest_test_author",
+          "https://kofi.sexy/role": "postgrest_test_author",
           "id": "jdoe",
           "aud": "youraudience"
         }
 
     -}
-    let auth = authHeaderJWT "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjk5OTk5OTk5OTksInJvbGUiOiJwb3N0Z3Jlc3RfdGVzdF9hdXRob3IiLCJpZCI6Impkb2UiLCJhdWQiOiJ5b3VyYXVkaWVuY2UifQ.fJ4tLKSmolWGWehWN20qiU9dMO-WY0RI2VvacL7-ZGo"
+    let auth = authHeaderJWT "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjk5OTk5OTk5OTksImh0dHBzOi8va29maS5zZXh5L3JvbGUiOiJwb3N0Z3Jlc3RfdGVzdF9hdXRob3IiLCJpZCI6Impkb2UiLCJhdWQiOiJ5b3VyYXVkaWVuY2UifQ.M3A8agB2l7Etck9YZMm1OR3ql7-VakUBQc1pzYTNxIU"
     request methodGet "/authors_only" [auth] ""
       `shouldRespondWith` 200
 
@@ -39,9 +39,9 @@ spec = describe "test handling of aud claims in JWT" $ do
           "id": "jdoe"
         }
     -}
-    let auth = authHeaderJWT "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjk5OTk5OTk5OTksInJvbGUiOiJwb3N0Z3Jlc3RfdGVzdF9hdXRob3IiLCJpZCI6Impkb2UifQ.Dpss-QoLYjec5OTsOaAc3FNVsSjA89wACoV-0ra3ClA"
+    let auth = authHeaderJWT "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjk5OTk5OTk5OTksImh0dHBzOi8va29maS5zZXh5L3JvbGUiOiJwb3N0Z3Jlc3RfdGVzdF9hdXRob3IiLCJpZCI6Impkb2UifQ.PJLrszyJw5TClpuk-fU5LefyNzkPb_98u5VbdwZvgT8"
     request methodGet "/authors_only" [auth] ""
       `shouldRespondWith` 200
 
   it "requests without JWT token should work" $
-    get "/has_count_column" `shouldRespondWith` 200 
\ No newline at end of file
+    get "/has_count_column" `shouldRespondWith` 200 
diff --git a/test/Feature/QuerySpec.hs b/test/Feature/QuerySpec.hs
index 86b9a1a..25c00a9 100644
--- a/test/Feature/QuerySpec.hs
+++ b/test/Feature/QuerySpec.hs
@@ -769,7 +769,7 @@ spec = do
           }
     it "current role is available as GUC claim" $
       request methodPost "/rpc/get_guc_value" []
-          [json| { "name": "request.jwt.claim.role" } |]
+          [json| { "name": "request.jwt.claim.https://kofi.sexy/role" } |]
           `shouldRespondWith`
           [str|"postgrest_test_anonymous"|]
           { matchStatus  = 200
